File: tests/java/src/main/java/enum_/versions/Referenced_Conversions.java
package enum_.versions;

import dev.argon.verilization.runtime.Converter;

final class Referenced_Conversions {
    private Referenced_Conversions() {}


    static final Converter<Referenced.V1, Referenced.V2> v1ToV2 = new Converter<Referenced.V1, Referenced.V2>() {
        @Override
        public Referenced.V2 convert(Referenced.V1 prev) {
            return switch(prev) {
                case Referenced.V1.X prev2 -> new Referenced.V2.X(prev2.x());
            };
        }
    };
    

}

File: tests/scala/shared/src/main/scala/enum_/versions/Referenced_Conversions.scala
package enum_.versions

import dev.argon.verilization.scala_runtime.Converter

object Referenced_Conversions {
	val v1ToV2: Converter[Referenced.V1, Referenced.V2] = {
		case Referenced.V1.X(x) => Referenced.V2.X(x)
	}
}
